OpenRA.Game/Network/ReplayRecorderConnection.cs

```diff
@@ -89,10 +89,13 @@ public void Receive(Action<int, byte[]> packetFn)
 						StartSavingReplay(preStartData);
 					}
 
-					writer.Write(client);
-					writer.Write(data.Length);
-					writer.Write(data);
-					packetFn(client, data);
+					if (!disposed)
```

Comment:
I don't think its valid to do anything in here if its been disposed... how about adding an `if (disposed)\nreturn;` at the top instead?